### ISSUE TITLE
Dative bonds with s- and p-elements as acceptors

### ISSUE THREAD

**Issue by IvanChernyshov** [rdkit/rdkit#3360]
First of all, I was surprised that water complexes of alkaline and alkaline-earth metals can not be sanitized:

```python
m1 = Chem.MolFromSmiles('[Li+](<-O)(<-O)(<-O)<-O')
m2 = Chem.MolFromSmiles('[Ca+2](<-O)(<-O)(<-O)(<-O)(<-O)<-O')
m3 = Chem.MolFromSmiles('[Al+3](<-O)(<-O)(<-O)(<-O)(<-O)<-O')
if (m1 is None) and (m2 is None) and (m3 is None):
    print('Oops')
> Oops
```

After that, I found interesting dependence: ions of s-/p- elements can't form any dative bonds as acceptors:

```python
m1 = Chem.MolFromSmiles('[Li+]<-O')
m2 = Chem.MolFromSmiles('[Ca+2]<-O')
m3 = Chem.MolFromSmiles('[Al+3]<-O')
if (m1 is None) and (m2 is None) and (m3 is None):
    print('Oops')
> Oops
```

but neutral atoms can form up to {max valence} dative bonds:

```python
m1 = Chem.MolFromSmiles('[Li]<-O')
m2 = Chem.MolFromSmiles('[Ca](<-O)<-O')
m3 = Chem.MolFromSmiles('[Al](<-O)(<-O)<-O')
if m1 and m2 and m3:
    print('Wooow')
> Wooow
```

It means that dative bonds are treated as single bonds for s- and p- elements. That's is in line with that "dative bonds have the special characteristic that they don’t affect the valence on the start atom, but do affect the end atom", but there are two issues:

- an atomic charge is treated incorrectly;
- alkaline and alkaline-earth metals can form more than 8 dative bonds.

Summing up, I think it's good idea:

- to fix the incorrect charge treating, and
- to do so that dative bonds won't affect the valence of the acceptors for alkaline, alkaline-earth, and Al subgroup elements (for boron it must be treated as for carbocations).

**Comment by greglandrum** [comment]
@IvanChernyshov : often these things come down to a mismatch in the way structures are represented. Can you provide an example of a structure (ideally with a link to an experimental paper) demonstrating an example of an `[Li+]` with four dative bonds to it or a `[Ca2+]` with six dative bonds to it?

**Comment by IvanChernyshov** [comment]
@greglandrum Sure, and the easiest way to show it is to use CSD:
- ConQuest query: <img width="276" alt="Ca_6" src="https://user-images.githubusercontent.com/17598902/91649330-7c915e00-ea7b-11ea-84e8-95f59b60130e.png">
- ConQuest output: <img width="110" alt="Ca_6_res" src="https://user-images.githubusercontent.com/17598902/91649333-8b781080-ea7b-11ea-8791-5b2b71c855e2.png">
- Ref for the first output structure: https://onlinelibrary.wiley.com/doi/abs/10.1002/asia.201901801 (Scheme 5, structure 6; in addition, this article contains nice examples for Mg+2 and Li+)

As I can see now, this thread should be merged with the previous one (https://github.com/rdkit/rdkit/issues/3359).
